### PR TITLE
fix(dApp explorer): display SA banner on modal for  reconnecting dapps

### PR DESCRIPTION
**Pull request by rahulnr7**
##### Ticket

[YOMO-1841](https://emurgo.atlassian.net/browse/YOMO-1841)

> [!NOTE]
> Please create the ticket if missing it.

[YOMO-1841]: https://emurgo.atlassian.net/browse/YOMO-1841?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/wallet-mobile/src/features/Discover/useCases/SelectDappFromList/DAppListItem/DAppListItem.tsx [hunk 1]
```diff
@@ -42,7 +45,16 @@ export const DAppListItem = ({dApp, connected, onPress}: Props) => {
...
-  const heightDialogByHeightScreen = (HEIGHT_SCREEN * 40) / 100
+
+  // For devices like iPhone SE wuth logical pixel height < 700, we need a bottom sheet with more height
+  const heightDialogByHeightScreen = dApp.isSingleAddress
+    ? HEIGHT_SCREEN < 700
+      ? (HEIGHT_SCREEN * 70) / 100
+      : (HEIGHT_SCREEN * 50) / 100
+    : HEIGHT_SCREEN < 700
+    ? (HEIGHT_SCREEN * 50) / 100
+    : (HEIGHT_SCREEN * 40) / 100
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Discover/useCases/SelectDappFromList/DAppListItem/DAppListItem.tsx]: looks solid, yet I think for readability looks simpler
```suggestion
  const heightDialogByHeightScreen = HEIGHT_SCREEN < 700
    ? (HEIGHT_SCREEN * (dApp.isSingleAddress ? 70 : 50)) / 100
    : (HEIGHT_SCREEN * (dApp.isSingleAddress ? 50 : 40)) / 100
```

File: apps/wallet-mobile/src/features/Discover/useCases/SelectDappFromList/DAppListItem/DAppListItem.tsx [hunk 2]
```diff
@@ -229,6 +278,13 @@ const useStyles = () => {
       ...atoms.body_1_lg_medium,
       color: color.gray_c900,
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Discover/useCases/SelectDappFromList/DAppListItem/DAppListItem.tsx]: @rahulnr7 for text from now on it needs to use `el_text_gray_<normal|max|min>`

File: apps/wallet-mobile/src/features/Discover/useCases/SelectDappFromList/DAppListItem/DAppListItem.tsx [hunk 3]
```diff
@@ -229,6 +278,13 @@ const useStyles = () => {
...
     },
+    warningText: {
+      ...atoms.body_2_md_regular,
+      color: color.gray_cmax,
+    },
+    link: {
+      color: color.sys_cyan_c500,
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Discover/useCases/SelectDappFromList/DAppListItem/DAppListItem.tsx:283]: :arrow_up:

File: apps/wallet-mobile/src/features/Discover/useCases/SelectDappFromList/DAppListItem/DAppListItem.tsx [hunk 4]
```diff
@@ -229,6 +278,13 @@ const useStyles = () => {
       ...atoms.body_1_lg_medium,
       color: color.gray_c900,
     },
+    warningText: {
+      ...atoms.body_2_md_regular,
+      color: color.gray_cmax,
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Discover/useCases/SelectDappFromList/DAppListItem/DAppListItem.tsx]: :arrow_up:

**Review by stackchain — APPROVED** [review]
lgtm 2 points, waiting CR updates to merge it.

File: apps/wallet-mobile/src/features/Discover/useCases/SelectDappFromList/DAppListItem/DAppListItem.tsx (hunk 2 above)
Comment (rahulnr7) [reply apps/wallet-mobile/src/features/Discover/useCases/SelectDappFromList/DAppListItem/DAppListItem.tsx]: done

File: apps/wallet-mobile/src/features/Discover/useCases/SelectDappFromList/DAppListItem/DAppListItem.tsx (hunk 3 above)
Comment (rahulnr7) [reply apps/wallet-mobile/src/features/Discover/useCases/SelectDappFromList/DAppListItem/DAppListItem.tsx:283]: done

**Pull request merged**